### PR TITLE
fix: support wait operators close

### PR DESCRIPTION
**Pull request by Yisaer**

### CODE REVIEW

File: contract/api/ctx.go [hunk 1]
```diff
@@ -45,6 +45,7 @@ type StreamContext interface {
 	GetOpId() string
 	GetInstanceId() int
 	GetRootPath() string
+	GetRuleWaitGroup() *sync.WaitGroup
```
Comment (ngjaying) [inline contract/api/ctx.go:48]: Looks like this is an internal signal, do not need to export here? Just pass into node when creating topo/planning.

**Review by ngjaying — CHANGES_REQUESTED** [review]

File: internal/conf/conf.go [hunk 2]
```diff
@@ -356,6 +357,10 @@ func InitConf() {
 		}
 	}
 
+	if time.Duration(Config.Basic.GracefulShutdownTimeout) < 1 {
```
[the remark this answers is not in the document]
Comment (Yisaer) [reply internal/conf/conf.go:360]: it's expected.

File: contract/api/ctx.go (hunk 1 above)
Comment (Yisaer) [reply contract/api/ctx.go]: updated.

**Review by ngjaying — APPROVED** [review]

**Comment by ngjaying** [comment]
Closes #2525

**Pull request merged**